src/Type/MediaItem/MediaItemCreate.php

```diff
@@ -0,0 +1,195 @@
+	 *
+	 * @var \WP_Post_Type $post_type_object
+	 *
+	 * @return array|mixed
+	 */
+	public static function mutate( \WP_Post_Type $post_type_object ) {
+
+		/**
+		 * Set the name of the mutation being performed
+		 */
+		$mutation_name = 'createMediaItem';
+
+		self::$mutation['mediaItem'] = Relay::mutationWithClientMutationId( [
+			'name' => esc_html( $mutation_name ),
+			'description' => __( 'Create mediaItems', 'wp-graphql' ),
+			'inputFields' => self::input_fields( $post_type_object ),
+			'outputFields' => [
+				'mediaItem' => [
+					'type' => Types::post_object( $post_type_object->name ),
+					'resolve' => function( $payload ) {
+						return get_post( $payload['id'] );
+					},
+				],
+			],
+			'mutateAndGetPayload' => function( $input ) use ( $post_type_object, $mutation_name ) {
+
+				/**
+				 * Throw an exception if there's no input
+				 */
+				if ( ( empty( $post_type_object->name ) ) || ( empty( $input ) || ! is_array( $input ) ) ) {
+					throw new \Exception( __( 'Mutation not processed. There was no input for the mutation or the post_type_object was invalid', 'wp-graphql' ) );
+				}
+
+				/**
+				 * Stop now if a user isn't allowed to create a mediaItem
+				 */
+				if ( ! current_user_can( $post_type_object->cap->create_posts ) ) {
+					throw new \Exception( __( 'Sorry, you are not allowed to create mediaItems', 'wp-graphql' ) );
+				}
+
+				/**
+				 * If the mediaItem being created is being assigned to another user that's not the current user, make sure
+				 * the current user has permission to edit others mediaItems
+				 */
+				if ( ! empty( $input['authorId'] ) && get_current_user_id() !== $input['authorId'] && ! current_user_can( $post_type_object->cap->edit_others_posts ) ) {
+					throw new \Exception( __( 'Sorry, you are not allowed to create mediaItems as this user', 'wp-graphql' ) );
+				}
+
+				/**
+				 * Set the file name, whether it's a local file or from a URL
+				 */
+				$file_name = basename( $input['filePath'] );
+
+				/**
+				 * Check if the download_url method exists and include it if not
+				 * This file also includes the wp_handle_sideload method
+				 */
+				if ( ! function_exists( 'download_url' ) ) {
+					require_once( ABSPATH . 'wp-admin/includes/file.php' );
+				}
+
+				/**
+				 * If the mediaItem file is from a local server, use wp_upload_bits before saving it to the uploads folder
+				 */
+				if ( 'false' === filter_var( $input['filePath'], FILTER_VALIDATE_URL ) ) {
+					$uploaded_file = wp_upload_bits( $file_name, null, file_get_contents( $input['filePath'] ) );
+					$uploaded_file_url = $uploaded_file['url'];
+				} else {
+					$uploaded_file_url = $input['filePath'];
+				}
+
+				/**
+				 * URL data for the mediaItem, timeout value is the default, see:
+				 * https://developer.wordpress.org/reference/functions/download_url/
+				 */
+				$timeout_seconds = 300;
+				$temp_file = download_url( $uploaded_file_url, $timeout_seconds );
```

Comment:
Since this function can return a WP_Error, we should account for that and throw an exception when that happens.

https://codex.wordpress.org/Function_Reference/download_url